cfme/common/host_views.py

```diff
@@ -0,0 +1,365 @@
+# -*- coding: utf-8 -*-
+from widgetastic.widget import ParametrizedView, Text, View
+from widgetastic_patternfly import (
+    BootstrapSelect,
+    CheckableBootstrapTreeview,
+    Dropdown,
+    FlashMessages,
+    Tab
+)
+from widgetastic.utils import ParametrizedLocator
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound, ManyItemsFound
+from widgetastic_manageiq import (
+    BaseItem,
+    BaseListItem,
+    BaseQuadIconItem,
+    BaseTileIconItem,
+    BootstrapTreeview,
+    BreadCrumb,
+    Button,
+    Checkbox,
+    Input,
+    ItemsToolBarViewSelector,
+    PaginationPane,
+    Search,
+    SummaryTable,
+    Table,
+    TimelinesView
+)
+
+
+class ComputeInfrastructureHostsView(BaseLoggedInPage):
+    """Common parts for host views."""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]'
+    )
+
+    @property
+    def in_compute_infrastructure_hosts(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ["Compute", "Infrastructure", "Hosts"]
+        )
+
+
+class HostQuadIconItem(BaseQuadIconItem):
+
+    @property
```

Comment:
I see. But It makes sense to follow the same approach everywhere. 
I had to create data property there because f.e. list items have much more data than quad icons and we also need to make that data accessable. So, I made one dict property for all cases